docs/cookbook/extending-product-list.md

```diff
@@ -66,9 +66,116 @@ class ListedProductView extends BaseListedProductView
+            ->setMaxResults($batchSize);
+
+        $query = $queryBuilder->getQuery();
+
+        $result = [];
+        /** @var \Shopsys\FrameworkBundle\Model\Product\Product $product */
+        foreach ($query->getResult() as $product) {
+            $flagIds = $this->extractFlags($product);
+            $categoryIds = $this->extractCategories($domainId, $product);
+            $parameters = $this->extractParameters($locale, $product);
+            $prices = $this->extractPrices($domainId, $product);
+            $visibility = $this->extractVisibility($domainId, $product);
+
+            $friendlyUrl = $this->friendlyUrlRepository->getMainFriendlyUrl($domainId, 'front_product_detail', $product->getId());
+
+            $result[] = [
+                'id' => $product->getId(),
+                'catnum' => $product->getCatnum(),
+                'partno' => $product->getPartno(),
+                'ean' => $product->getEan(),
+                'name' => $product->getName($locale),
+                'description' => $product->getDescription($domainId),
+                'shortDescription' => $product->getShortDescription($domainId),
+                'brand' => $product->getBrand() ? $product->getBrand()->getId() : '',
+                'brand_name' => $product->getBrand() ? $product->getBrand()->getName() : '',
+                'flags' => $flagIds,
+                'categories' => $categoryIds,
+                'in_stock' => $product->getCalculatedAvailability()->getDispatchTime() === 0,
+                'prices' => $prices,
+                'parameters' => $parameters,
+                'ordering_priority' => $product->getOrderingPriority(),
+                'calculated_selling_denied' => $product->getCalculatedSellingDenied(),
+                'selling_denied' => $product->isSellingDenied(),
+                'availability' => $product->getAvailability()->getName($locale),
+                'main_variant' => $product->isMainVariant(),
+                'detail_url' => $friendlyUrl->getAbsoluteUrl($this->domain),
+                'visibility' => $visibility,
+            ];
+        }
+
+        return $result;
+    }
+}
+```
+
+You need to register your new class as an alias for the one from the FrameworkBundle in `services.yml`:
+
+```yml
+Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository: '@Shopsys\ShopBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository'
+```
+
+Then you need to fix `ProductSearchExportRepositoryTest::getExpectedStructureForRepository` (because this test check if your structure is correct) by adding new attribute:
+```diff
+$structure = \array_merge($structure, [
+    'availability',
+    'brand',
+    'flags',
+    'categories',
+    'detail_url',
+    'in_stock',
+    'prices',
+    'parameters',
+    'ordering_priority',
+    'calculated_selling_denied',
+    'selling_denied',
+    'main_variant',
+    'visibility',
++   'brand_name',
+]);
+```
```

Comment:
When I finished the cookbook, my tests start to fail on following errors (i have ext_id instead of brand_name):

```
1) Tests\ReadModelBundle\Functional\Product\Listed\ListedProductViewFacadeTest::testGetPaginatedForBrand
Symfony\Component\Debug\Exception\ContextErrorException: Notice: Undefined index: ext_id

/Volumes/work/shopsys/framework/project-base/src/Shopsys/ShopBundle/ReadModel/ListedProductViewFactory.php:34
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:241
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:222
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:209
/Volumes/work/shopsys/framework/project-base/tests/ReadModelBundle/Functional/Product/Listed/ListedProductViewFacadeTest.php:43

2) Tests\ReadModelBundle\Functional\Product\Listed\ListedProductViewFacadeTest::testGetFilteredPaginatedForSearch
Symfony\Component\Debug\Exception\ContextErrorException: Notice: Undefined index: ext_id

/Volumes/work/shopsys/framework/project-base/src/Shopsys/ShopBundle/ReadModel/ListedProductViewFactory.php:34
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:241
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:222
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:195
/Volumes/work/shopsys/framework/project-base/tests/ReadModelBundle/Functional/Product/Listed/ListedProductViewFacadeTest.php:57

3) Tests\ReadModelBundle\Functional\Product\Listed\ListedProductViewFacadeTest::testGetFilteredPaginatedInCategory
Symfony\Component\Debug\Exception\ContextErrorException: Notice: Undefined index: ext_id

/Volumes/work/shopsys/framework/project-base/src/Shopsys/ShopBundle/ReadModel/ListedProductViewFactory.php:34
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:241
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:222
/Volumes/work/shopsys/framework/packages/read-model/src/Product/Listed/ListedProductViewFacade.php:180
/Volumes/work/shopsys/framework/project-base/tests/ReadModelBundle/Functional/Product/Listed/ListedProductViewFacadeTest.php:90

4) Tests\ShopBundle\Functional\Controller\ProductRenameRedirectPreviousUrlTest::testPreviousUrlRedirect
Symfony\Component\Debug\Exception\ContextErrorException: Notice: Undefined property: Shopsys\FrameworkBundle\Model\Product\ProductData::$extId

/Volumes/work/shopsys/framework/project-base/src/Shopsys/ShopBundle/Model/Product/Product.php:47
/Volumes/work/shopsys/framework/packages/framework/src/Model/Product/ProductFacade.php:267
/Volumes/work/shopsys/framework/project-base/tests/ShopBundle/Functional/Controller/ProductRenameRedirectPreviousUrlTest.php:33

```

Reply:
I don't think that this is because of this cookbook. Are you sure, you didn't forget to add something?